exe/cli/cmd/root.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	client "github.com/textileio/filecoin/api/stub"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	cfgFile string
+
+	fcClient *client.Client
+
+	cmdTimeout = time.Second * 10
+
+	rootCmd = &cobra.Command{
+		Use:   "filecoin",
+		Short: "A client for storage and retreival of filecoin data",
+		Long:  `A client for storage and retreival of filecoin data`,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			err := viper.BindPFlag("serverAddress", cmd.Root().PersistentFlags().Lookup("serverAddress"))
+			checkErr(err)
+
+			target := viper.GetString("serverAddress")
+			var creds credentials.TransportCredentials
+			if strings.Contains(target, "443") {
+				creds = credentials.NewTLS(&tls.Config{})
+			}
+			fcClient, err = client.NewClient(target, creds)
+			checkErr(err)
+		},
+	}
+)
+
+// Execute runs the root command
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.filecoin.yaml)")
+	rootCmd.PersistentFlags().String("serverAddress", "127.0.0.1:5002", "address of the filecoin service api")
+}
+
+func initConfig() {
```

Comment:
Dup from helpers?

Reply:
Yep. Will de-dup.

Reply:
For now I just went with the existing code that is being used. It's simple. Removed all the duplicated, unused code here and made an issue to look at implementing a more robust config file location mechanism in the future if we want https://github.com/textileio/fil-tools/issues/72